### PR TITLE
Fix Flutter Tools Tests to run on Windows

### PR DESCRIPTION
**Pull request by goderbauer**

### CODE REVIEW

File: packages/flutter_tools/lib/src/dart/analysis.dart [hunk 1]
```diff
@@ -46,7 +46,9 @@ class AnalysisDriver {
 
   DriverOptions options;
 
-  String get sdkDir => options.dartSdkPath ?? cli_util.getSdkDir().path;
+  String get sdkDir {
+    return options.dartSdkPath ?? path.absolute(cli_util.getSdkDir().path);
```
Comment (eseidelGoogle) [inline packages/flutter_tools/lib/src/dart/analysis.dart]: So this is the only command which respects dartSdkPath?  Odd that format above doesn't for example.

File: packages/flutter_tools/lib/src/commands/format.dart [hunk 2]
```diff
@@ -37,8 +38,9 @@ class FormatCommand extends FlutterCommand {
       );
     }
 
+    String executable = os.getExecutableName('dartfmt', winExtension: 'bat');
     String dartfmt = path.join(
-        Cache.flutterRoot, 'bin', 'cache', 'dart-sdk', 'bin', 'dartfmt');
+        Cache.flutterRoot, 'bin', 'cache', 'dart-sdk', 'bin', executable);
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/commands/format.dart]: nit: newline before the `)`

Comment (Hixie) [inline packages/flutter_tools/lib/src/commands/format.dart]: or just put it all on one line

File: packages/flutter_tools/test/src/context.dart [hunk 3]
```diff
@@ -35,7 +35,8 @@ typedef dynamic Generator();
 
 void testUsingContext(String description, dynamic testMethod(), {
   Timeout timeout,
-  Map<Type, Generator> overrides: const <Type, Generator>{}
+  Map<Type, Generator> overrides: const <Type, Generator>{},
+  bool skip: false
```
Comment (Hixie) [inline packages/flutter_tools/test/src/context.dart]: nit: trailing comma

**Comment by Hixie** [comment]
LGTM. I wish there was a way we could check this in an automated fashion, but...

File: packages/flutter_tools/lib/src/dart/analysis.dart (hunk 1 above)
Comment (goderbauer) [inline packages/flutter_tools/lib/src/dart/analysis.dart]: There are other commands that use the dart SDK (e.g. format calls the dartfmt binary directly), but this seems to be the only one that uses internals of the Dart SDK, hence the slightly different call.

**Comment by tvolkert** [comment]
LGTM.  Yeah, I'm following https://github.com/travis-ci/travis-ci/issues/216 for when Travis adds Windows support.

**Comment by goderbauer** [comment]
Running these tests on Travis would be awesome. In the meantime I am working on getting them running on our Windows Builtbots.

**Pull request merged**